File: my-solutions/0747-min-cost-climbing-stairs/solution.cpp
class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {
        int n = cost.size();
        int prev = cost[0];
        int curr = cost[1];

        for (int i = 2; i < n; ++i) {
            int res = cost[i] + min(prev, curr);
            prev = curr;
            curr = res;
        }

        return min(prev, curr);
    }
};

File: my-solutions/0747-min-cost-climbing-stairs/solution.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        dp = [0] * (len(cost) + 1)
        dp[-2] = cost[-1]

        for i in range(len(dp) - 3, -1, -1):
            dp[i] = min(dp[i + 1], dp[i + 2]) + cost[i]
        
        return min(dp[0], dp[1])
